feat: implement `CairoLayout`

tcoratger opened the pull request:
Should close #228.

tcoratger commented:
@lana-shanghai As this is `CairoLayout` and it is possible, in unit tests, to check each parameter in a hardcoded way, I did it. This is obviously a lot of lines of code but this way we are 100% sure and we will always see any changes. Don't hesitate to tell me if it's too many tests I can refactor, but I said to myself that I might as well test everything.

lana-shanghai commented on src/lib.zig:
```diff
@@ -13,6 +13,9 @@ pub const vm = struct {
     pub usingnamespace @import("vm/types/builtins_instance_def.zig");
+    pub usingnamespace @import("vm/types/diluted_pool_instance_def.zig");
+    pub usingnamespace @import("vm/types/cpu_instance_def.zig");
+    pub usingnamespace @import("vm/types/layout.zig");
```
Let's make sure we finish implementing the builtins before we touch the layout topic

lana-shanghai commented on src/vm/types/layout.zig:
```diff
@@ -1,8 +1,946 @@
+    /// - `memory_units_per_step`: 8
+    /// - `n_trace_columns`: 22
+    /// - Other instances set to default or null.
+    pub fn dexInstance() Self {
```
I think these instance definitions are redundant at the moment, let's discuss the ones we need after the builtins for the milestone. We also have to establish a commit which marks the end of the first milestone.

tcoratger commented:
@lana-shanghai FYI I just removed the unnecessary layouts leaving only the following layouts:
- Plain
- Small
- All Cairo
- Dynamic (I added it in addition to those that we mentioned yesterday because it is a configurable layout which could be useful very soon in my opinion)

lana-shanghai reviewed (approved):
lgtm
